model-optimizer/extensions/ops/If.py

```diff
@@ -0,0 +1,327 @@
+# Copyright (C) 2018-2021 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+import logging as log
+import numpy as np
+
+from mo.front.common.partial_infer.utils import int64_array
+from mo.graph.graph import Node, Graph
+from mo.middle.passes.infer import partial_infer
+from mo.ops.op import Op
+
+
+class If(Op):
+    """
+    If operation is an operation which has an input with condition which defines what sub-graph "then" or "else" to be
+    executed.
+    """
+    op = 'If'
+    enabled = False
+
+    def __init__(self, graph: Graph, attrs: dict):
+        base_attrs = {
+            'type': self.op,
+            'op': self.op,
+            'then_graph': None,  # an Graph object with a "then" body sub-graph (condition is True)
+            'else_graph': None,  # an Graph object with a "else" body sub-graph (condition is False)
+            'sub_graphs': ['then_graph', 'else_graph'],  # built-in attribute with all sub-graphs
+            'version': 'opset8',
+            'infer': self.infer,
+            'type_infer': self.type_infer,
+        }
+        base_attrs.update(attrs)
+        super().__init__(graph, base_attrs, attrs)
+
+    def port_map_attrs(self):
+        return [
+            'external_port_id',
+            'internal_layer_id'
+        ]
+
+    @staticmethod
+    def connect_body_input(if_node: Node, condition: bool, if_input_port_idx: int, body_parameter: Node):
+        """
+        Update the input port map to connect the input port with the specified body parameter
+
+        :param if_node: the If node
+        :param condition: the boolean defining a condition (then/else) graph to add connect the body
+        :param if_input_port_idx: the input port index to connect
+        :param body_parameter: the body parameter node to connect
+        :return: None
+        """
+        assert if_node.soft_get('op') == 'If'
+        assert body_parameter.soft_get('op') == 'Parameter'
+        sub_graph = if_node.then_graph if condition else if_node.else_graph
+        assert body_parameter.id in sub_graph
+        body_parameter['input_id'] = if_input_port_idx
```

Comment:
but actually here you don't update port map. Please double-check this function

Reply:
I dont use portmap in MO part, portmap is created before IR.